src/fs.cpp

```diff
@@ -31,6 +32,14 @@ FILE *fopen(const fs::path& p, const char *mode)
 #endif
 }
 
+fs::path unique_path() {
+    // TODO: Do not use Boost in this implementation
+    boost::filesystem::path p = boost::filesystem::unique_path();
+    fs::path p2{p.string()};
+
+    return p2;
```

Comment:
I think all that boost does it pick 64 bits of randomness? Something along the lines of 

```suggestion
    return g_insecure_rand_ctx_temp_path.rand256().ToString().substr(0, 16);
```

should be identical.

Reply:
Yes. I believe so too.

However, using this leads to a new circular dependency "fs -> random -> logging -> fs": https://travis-ci.com/github/kiminuo/bitcoin/jobs/360282950#L466 (https://github.com/kiminuo/bitcoin/commit/333ab6ad421f1b25bf66bd2557cd3d0e4b60e0f6)

`unique_path()` is used only twice but I don't know what is preferred solution here. Inline it? Move function somewhere else? Where?

_____
For other people who may read this: For [used](https://github.com/bitcoin/bitcoin/blob/master/doc/dependencies.md) Boost 1.70.0:

* Source code: https://github.com/boostorg/filesystem/blob/e49775e8bbc4d8dbea5fbf6cf7c29f4d7bbd0c67/src/unique_path.cpp#L131
* Docs: https://www.boost.org/doc/libs/1_70_0/libs/filesystem/doc/reference.html#unique_path

Reply:
I think there are two options:

* Inline
* or create a new header, e.g. `src/util/fs.h`